test/sql-luatest/triggers_test.lua

```diff
@@ -0,0 +1,684 @@
+
+        exp = {
+            metadata = {
+                {
+                    name = "x",
+                    type = "integer",
+                },
+            },
+            rows = {{1}, {2}, {3}},
+        }
+        t.assert_equals(box.execute("SELECT * FROM t2;"), exp)
+        box.execute("DELETE FROM t2;")
+
+        -- Test t1t after t2t and t3t drop.
+        box.execute("DROP TRIGGER t2t;")
+        _trigger:delete("t3t")
+        sql = "CREATE TRIGGER t1t AFTER INSERT ON t1 FOR EACH ROW BEGIN "..
+              "INSERT INTO t2 VALUES(1); END;"
+        check_triggers_def({{"t1t", {sql = sql}}})
+        box.execute("INSERT INTO t1 VALUES(3);")
+
+        exp = {
+            metadata = {
+                {
+                    name = "x",
+                    type = "integer",
+                },
+            },
+            rows = {{1}},
+        }
+        t.assert_equals(box.execute("SELECT * FROM t2;"), exp)
+        box.execute("DELETE FROM t2;")
+
+        -- Insert new SQL t2t and t3t.
+        sql = "CREATE TRIGGER t2t AFTER INSERT ON t1 FOR EACH ROW BEGIN "..
+              "INSERT INTO t2 VALUES(2); END;"
+        box.execute(sql)
+        sql = "CREATE TRIGGER t3t AFTER INSERT ON t1 FOR EACH ROW BEGIN "..
+              "INSERT INTO t2 VALUES(3); END;"
+        box.execute(sql)
+        exp = {
+            {
+                "t1t",
+                {
+                    sql = "CREATE TRIGGER t1t AFTER INSERT ON t1 FOR "..
+                          "EACH ROW BEGIN INSERT INTO t2 VALUES(1); END;",
+                },
+            },
+            {
+                "t2t",
+                {
+                    sql = "CREATE TRIGGER t2t AFTER INSERT ON t1 FOR "..
+                          "EACH ROW BEGIN INSERT INTO t2 VALUES(2); END;",
+                },
+            },
+            {
+                "t3t",
+                {
+                    sql = "CREATE TRIGGER t3t AFTER INSERT ON t1 FOR "..
+                          "EACH ROW BEGIN INSERT INTO t2 VALUES(3); END;",
+                },
+            },
+        }
+        check_triggers_def(exp)
+        box.execute("INSERT INTO t1 VALUES(4);")
+
+        exp = {
+            metadata = {
+                {
+                    name = "x",
+                    type = "integer",
+                },
+            },
+            rows = {{1}, {2}, {3}},
+        }
+        t.assert_equals(box.execute("SELECT * FROM t2;"), exp)
+
+        -- Clean up.
+        box.execute("DROP TABLE t1;")
+        t.assert_equals(box.space.t1, nil)
+        box.execute("DROP TABLE t2;")
+        t.assert_equals(box.space.t2, nil)
+        check_triggers_def({})
+
+        -- Test target tables restricts.
+        box.execute("CREATE TABLE t1(a INT PRIMARY KEY,b INT);")
+        space_id = box.space.t1.id
+
+        sql = [[CREATE TRIGGER t1t INSTEAD OF UPDATE ON
+                t1 FOR EACH ROW BEGIN DELETE FROM t1 WHERE
+                a=old.a+2; END;]]
```

Comment:
Let's refactor this line a bit:
```Lua
        sql = [[CREATE TRIGGER t1t INSTEAD OF UPDATE ON t1 FOR EACH ROW
                BEGIN DELETE FROM t1 WHERE a = old.a + 2; END;]]
```
Notice that I added some spaces around `=` and `+`.